macaw-cli/src/Data/Macaw/CLI.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Data.Macaw.CLI
+  ( sim
+  , ppSimRes
+  ) where
+
+import Control.Lens qualified as Lens
+import Data.ByteString.Char8 qualified as BS8
+import Data.List qualified as List
+import Data.Map qualified as Map 
+import Data.Text qualified as Text 
+import GHC.TypeLits (KnownNat)
+
+-- First-party
+import Data.ElfEdit qualified as Elf
+import Data.Macaw.Architecture.Info qualified as MAI
+import Data.Macaw.CFG qualified as MCFG
+import Data.Macaw.CLI.Options qualified as MCO
+import Data.Macaw.Discovery qualified as MD
+import Data.Macaw.Memory.ElfLoader.PLTStubs qualified as MPLT
+import Data.Macaw.Symbolic qualified as MS
+import Data.Macaw.Symbolic.Testing qualified as MST
```

Comment:
We are now using `Data.Macaw.Symbolic.Testing` for reasons beyond just simple testing. I wonder if we should consider migrating the name to something more appropriate. (`Data.Macaw.Symbolic.Driver`, perhaps?)

Reply:
I've also observed that `Data.Macaw.Symbolic.Testing` has some peculiar conventions that we may not want to use in `macaw-cli`. For instance, `Data.Macaw.Symbolic.Testing` won't take assumptions into account when checking an entrypoint function [unless it begins with the prefix "`test_and_verify_`"](https://github.com/GaloisInc/macaw/blob/645754178a034c22e97aac49a962c99b387aaa01/symbolic/src/Data/Macaw/Symbolic/Testing.hs#L433), so most functions won't check if they use the memory model in valid ways. This seems undesirable for `macaw-cli`'s needs.

Perhaps we could factor out the code used in common between `Data.Macaw.Symbolic.Testing` and the new module we use to power `macaw-cli` (be it named `Data.Macaw.Symbolic.Driver` or something else).